src/jdk.jdeps/share/classes/com/sun/tools/jdeps/ClassFileReader.java

```diff
@@ -217,12 +220,22 @@ public String toString() {
         }
 
         protected Set<String> scan() {
-            try (Stream<Path> stream = Files.walk(path, Integer.MAX_VALUE)) {
-                return stream.filter(ClassFileReader::isClass)
-                             .map(path::relativize)
-                             .map(Path::toString)
-                             .map(p -> p.replace(File.separatorChar, '/'))
-                             .collect(Collectors.toSet());
+            try {
+                Set<String> result = new HashSet<>();
+                Files.walkFileTree(path, new SimpleFileVisitor<>() {
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        if (ClassFileReader.isClass(file)) {
+                            result.add(path.relativize(file).toString().replace(File.separatorChar, '/'));
+                        }
+                        return FileVisitResult.CONTINUE;
+                    }
+                    @Override
+                    public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                        return FileVisitResult.CONTINUE;
```

Comment:
The bug report may be pilot error in that they seem to have specified a root directory as the class path. If the scan is changed then maybe it should minimally emit a warning rather than silently ignoring the non-accessible files.